cdc/entry/mounter.go

```diff
@@ -386,52 +395,90 @@ func (m *mounterImpl) mountRowKVEntry(tableInfo *TableInfo, row *rowKVEntry) (*m
 			whereHandle := true
 			col.WhereHandle = &whereHandle
 		}
-		values[colName] = col
+		cols[colName] = col
+	}
+	if !fillWithDefaultValue {
+		return cols, nil
+	}
+	for _, col := range tableInfo.Columns {
+		_, ok := cols[col.Name.O]
+		if !ok && tableInfo.IsColWritable(col) {
+			column := &model.Column{
+				Type:  col.Tp,
+				Value: getDefaultOrZeroValue(col),
+				Flag:  transColumnFlag(col),
+			}
+			if tableInfo.IsColumnUnique(col.ID) {
+				whereHandle := true
+				column.WhereHandle = &whereHandle
+			}
+			cols[col.Name.O] = column
+		}
+	}
+	return cols, nil
+}
+
+func (m *mounterImpl) mountRowKVEntry(tableInfo *TableInfo, row *rowKVEntry) (*model.RowChangedEvent, error) {
+	// if m.enableOldValue == true, go into this function
+	// if m.enableNewValue == false and row.Delete == false, go into this function
+	// if m.enableNewValue == false and row.Delete == true and tableInfo.PKIsHandle = true, go into this function
+	// only if m.enableNewValue == false and row.Delete == true and tableInfo.PKIsHandle == false, skip this function
+	if !m.enableOldValue && row.Delete && !tableInfo.PKIsHandle {
+		return nil, nil
+	}
+
+	var err error
+	// Decode previous columns.
+	var preCols map[string]*model.Column
+	if len(row.PreRow) != 0 {
+		// FIXME(leoppro): using pre table info to mounter pre column datum
+		// the pre column and current column in one event may using different table info
+		preCols, err = datum2Column(tableInfo, row.PreRow, true)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+	}
+
+	var cols map[string]*model.Column
+	disableOldValueAndDelete := !m.enableOldValue && row.Delete
```

Comment:
```suggestion
	oldValueDisabledAndRowIsDelete := !m.enableOldValue && row.Delete
```
Double negative is a little tweaked here 😂